corehq/apps/locations/models.py

```diff
@@ -681,22 +676,11 @@ def save(self, *args, **kwargs):
             ]
             self.site_code = generate_code(self.name, all_codes)
 
-        sql_location = None
-        result = super(Location, self).save(*args, **kwargs)
-
-        # try sync locations and when SQLLocation doesn't returned, removed Couch object from database.
-        # added because when we sync location by tasks we can have behavior that the task can be
-        # killed in _sync_location method and this causes the problems
-        try:
-            sql_location = self._sync_location()
-        finally:
-            if sql_location:
-                sql_location.save()
-            else:
-                self.delete()
-                result = None
-
-        return result
+        sync_to_sql = kwargs.pop('sync_to_sql', True)
+        kwargs['sync_to_sql'] = False  # only sync here
+        super(Location, self).save(*args, **kwargs)
+        if sync_to_sql:
+            self._migration_do_sync()
```

Comment:
for the messaging migrations i prefer to not let exceptions raise from the sync operations and instead notify them so that the sync process doesn't interfere with normal operations.  i'm not saying this is wrong here, but just something to give some thought to. i saw the sql to couch sync also raises exceptions.